src/preprocessor.rs

```diff
@@ -69,6 +69,7 @@ pub fn lookup(name: &str) -> Result<Box<dyn Preprocessor>> {
         "ingest-ns" => Ok(Box::new(ExtractIngresTs {})),
         "length-prefixed" => Ok(Box::new(LengthPrefix::default())),
         "textual-length-prefix" => Ok(Box::new(TextualLength::default())),
+        "zstd" => Ok(Box::new(Zstd::default())),
```

Comment:
We should include this in the `decompress` preprocessor for autodetecting zstd compressed data.

Reply:
LGTM - Adding fingerprinting to `decompress` is the last piece of the puzzle - it's very useful when the decompression algorithm can vary. Great work!